### PR TITLE
Added unit tests in athena-datalakegen2.

### PR DESCRIPTION
**Pull request by ritiktrianz**
*Issue #, if available:*

*Description of changes:*
Added unit tests in athena-datalakegen2.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+public class DataLakeGen2EnvironmentPropertiesTest
+{
+    private Map<String, String> connectionProperties;
+    private DataLakeGen2EnvironmentProperties dataLakeGen2EnvironmentProperties;
+
+    @Before
+    public void setUp()
+    {
+        connectionProperties = new HashMap<>();
+        connectionProperties.put(HOST, "afq-connection-000.sql.azuressss.net");
```
Comment (fal-bharadwaj) [inline athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java]: Is there a typo in the value here "azuressss"?

File: athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    public void connectionPropertiesToEnvironment_WithValidProperties_ReturnsExpectedConnectionString()
+    {
+        Map<String, String> dataLakeGen2ConnectionProperties = dataLakeGen2EnvironmentProperties.connectionPropertiesToEnvironment(connectionProperties);
+
+        String expectedConnectionString = "datalakegentwo://jdbc:sqlserver://afq-connection-000.sql.azuressss.net:1433;databaseName=testDB;${secret}";
+        assertEquals(expectedConnectionString, dataLakeGen2ConnectionProperties.get(DEFAULT));
+    }
+
+    @Test
+    public void getDelimiter_ReturnsSemicolonDelimiter()
```
Comment (fal-bharadwaj) [inline athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java]: Let's keep the test names consistent, add a "WithDefaultParams" or something similar.

File: athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Test
+    public void getConnectionStringPrefix_WithValidProperties_ReturnsPrefix()
+    {
+        assertEquals("datalakegentwo://jdbc:sqlserver://", dataLakeGen2EnvironmentProperties.getConnectionStringPrefix(connectionProperties));
+    }
+
+    @Test
+    public void getJdbcParametersSeparator_ReturnsSemicolonSeparator()
```
Comment (fal-bharadwaj) [inline athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java]: Missing tests for edge cases, null, empty values for required properties.

File: athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2MetadataHandlerTest.java [hunk 4]
```diff
@@ -428,4 +459,159 @@ public void doListTables() throws Exception
...
+        assertEquals(1, getSplitsResponse.getSplits().size());
+        Split split = getSplitsResponse.getSplits().iterator().next();
+        Map<String, String> properties = split.getProperties();
+        assertTrue("Split properties should contain query passthrough indicator", properties.containsKey("queryPassthrough"));
+        assertEquals("true", properties.get("queryPassthrough"));
+    }
+
+
+    @Test(expected = RuntimeException.class)
+    public void doGetTable_DataTypeQueryError_ThrowsRuntimeException() throws Exception {
```
Comment (fal-bharadwaj) [inline athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2MetadataHandlerTest.java:574]: Can we also test the exception message n code throw here?

**Review by fal-bharadwaj — CHANGES_REQUESTED** [review]

File: athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java (hunk 1 above)
Comment (ritiktrianz) [reply athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java]: Hi @fal-bharadwaj , thanks for the review. The value has been updated.

File: athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java (hunk 2 above)
Comment (ritiktrianz) [reply athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java]: Hi @fal-bharadwaj  , renamed test name to "getDelimiter_WithDefaultConfiguration_ReturnsSemicolonDelimiter()".

File: athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java (hunk 3 above)
Comment (ritiktrianz) [reply athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2EnvironmentPropertiesTest.java]: Added more tests to cover these scenarios.

File: athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2MetadataHandlerTest.java (hunk 4 above)
Comment (ritiktrianz) [reply athena-datalakegen2/src/test/java/com/amazonaws/athena/connectors/datalakegen2/DataLakeGen2MetadataHandlerTest.java:574]: Hi @fal-bharadwaj  , updated test to include these changes.

**Review by hareeshkumar-t — APPROVED** [review]
